rfc/plugins-ui.md

```diff
@@ -1,115 +1,124 @@
+Deck should provides a set of well defined extension points and expose them via the Plugin API object.  A number of extension points exist today.  Examples include:
 
-```
-element(document.documentElement).ready(() => {
-  bootstrap(document.documentElement, ['netflix.spinnaker']);
-});
-```
+- Stage Registry
+- Cloud provider Registry
+- Data Source Registry
+- Application tabs/routes
+- Component Overrides
+
+## Plugin Development
+
+Plugin developers should have a development experience that closely resembles development of Deck itself.  Spinnaker should provide a plugin quick-start, such as a github repository that developers can clone. This quick start should provide:
+
+- A standard build system
+- Templates for creating specific extension points, i.e., an example stage
+- A curated set of transitive dependencies (3rd party libraries) included in core Deck
+- A set of reusable UI components that spinnaker itself is built upon
+- A set of linter rules which match the code style in core Deck.
+
+The build system will ideally build a plugin as ES6 module(s).  Deck will import these ES6 modules using native browser `import()`.  The build system should ideally support code splitting to enable lazy loading of portions of plugin code.  The build system should not include the transitive dependencies that are already loaded by core Deck. Instead, it should share those curated dependencies at runtime from the existing code loaded into core deck.  
 
 ## Known Unknowns
-* How do we ensure end users can download plugin resources? 
-* Do we show a warning if we fail to reach Front50? 
-  * Spinnaker alert/banner at top of page should inform the user that plugins were not able to be loaded
-    * Only show this warning if plugins are enabled
-    * Give guidance on how to resolve error or look up more info
-* What if we fail to download a plugin resource? And how would we catch that (eg if its in a script tag)?
-  * plugin manifests will contain locations where assets are, but another place may serve them (eg secure artifact store), can halyard help with this?
+* Where are resources downloaded from? How do we ensure end users can download plugin resources?
+* Failure handling.  
+  * What should deck do if it fails to reach Front50 or to load plugins?
+    * Spinnaker alert/banner at top of page should inform the user that plugins were not able to be loaded
+      * Only show this warning if plugins are enabled
+      * Give guidance on how to resolve error or look up more info
+  * What should deck do if it fails to download a plugin resource? 
+* Plugin manifests will contain locations where assets are, but another place may serve them (eg secure artifact store), can halyard help with this?
```

Comment:
If plugin manifests are being served from front50, I think we could do some stuff on the way out of Gate to ensure locations are consistent (e.g. only show locations that are relative to gate, if that's the direction we end up going for serving plugin assets).